### DISCUSSION TITLE
Add a settings export file

### DISCUSSION THREAD

**Discussion by Strealstud** [arichornlover/uYouEnhanced#707] [Ideas]
As the title suggests please add a export settings option please if possible

**Comment by arichornlover** [comment]
It does already exist in uYouEnhanced but the option is hidden by default and I don’t plan on making it visible until I fix the problems with it.

**Comment by Strealstud** [comment]
Ah ok sorry didn’t know but I had an old version without the copy settings feature also maybe add a share feature that doesn’t require the app to use that exports the settings because if someone needs to re download the app but it keeps crashing they can’t export settings but also thanks for the app I love it

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by arichornlover** [comment]
Hey @Strealstud next release, the Export settings to a file will be possible!
by default the file will be called `exported_settings` so it should be easy to find!

https://github.com/arichornlover/uYouEnhanced/assets/78001398/d5a5c7e2-4bdb-43be-ac47-db08cabcc0c8

**Comment by Strealstud** [comment]
That’s good to know thanks but is it possible to export the settings by holding down on the app I have a video to show what I’m talking about
https://share.icloud.com/photos/0729cLybLXFVpGSjNULL5fK-w
Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by yodaluca23** [comment]
> That’s good to know thanks but is it possible to export the settings by holding down on the app I have a video to show what I’m talking about
> 
> https://share.icloud.com/photos/0729cLybLXFVpGSjNULL5fK-w

@Strealstud @arichornlover 

I think a better option would be every time the app is launched if it could automatically export the Settings file to the Files app in the YouTube app folder this would require editing the
.plist to allow the YouTube app to interact with the files app. This way if the certificate gets revoked or if you activate a setting that crashes the YouTube app on open, you have a backup of some of your settings still accessible, even if the YouTube app is completely inaccessible. Maybe store three back up setting files every time the app is launched check if the latest one is different then the latest one already in the folder, if it is then delete the oldest one and add the newest one. This way you always have the latest three backup files. This would be very complicated to implement I know, but it would be nice to have.

**Comment by Strealstud** [comment]
Wow that is pretty nice to have but as you said it would be very hard to implement and complicated I would offer to help but I’m not good with coding but also thanks for the app I love it

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by arichornlover** [comment]
> I think a better option would be every time the app is launched if it could automatically export the Settings file to the Files app in the YouTube app folder this would require editing the
.plist to allow the YouTube app to interact with the files app. This way if the certificate gets revoked or if you activate a setting that crashes the YouTube app on open, you have a backup of some of your settings still accessible, even if the YouTube app is completely inaccessible. Maybe store three back up setting files every time the app is launched check if the latest one is different then the latest one already in the folder, if it is then delete the oldest one and add the newest one. This way you always have the latest three backup files. This would be very complicated to implement I know, but it would be nice to have.

oh gosh, there’s no chance that I could succeed with this implementation, but this is a very good idea though.

**Reply by yodaluca23** [reply]
> > I think a better option would be every time the app is launched if it could automatically export the Settings file to the Files app in the YouTube app folder this would require editing the
> > .plist to allow the YouTube app to interact with the files app. This way if the certificate gets revoked or if you activate a setting that crashes the YouTube app on open, you have a backup of some of your settings still accessible, even if the YouTube app is completely inaccessible. Maybe store three back up setting files every time the app is launched check if the latest one is different then the latest one already in the folder, if it is then delete the oldest one and add the newest one. This way you always have the latest three backup files. This would be very complicated to implement I know, but it would be nice to have.
> 
> oh gosh, there’s no chance that I could succeed with this implementation, but this is a very good idea though.

Yea, I figured, maybe for now then just add it to the youtube folder every launch overriding anything their, not have the 3 latest thing.

**Comment by Strealstud** [comment]
Ah I see even with help? I’m sure there are people I could find or you could that could help

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Strealstud** [comment]
Also I am a iOS beta user and it works on the developer beta for iOS 18

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
